src/views/InteractiveMap/InteractiveMap.vue

```diff
@@ -457,59 +457,91 @@ export default Vue.extend({
+            if (newBound > 0) {
               return {
                 operation: "modify",
                 type: "reaction",
                 id: manipulation.id,
                 data: {
                   lower_bound: newBound,
-                  upper_bound: newBound
+                  upper_bound: model_reaction.upper_bound
                 }
               };
-            })
-            .filter(operation => operation.data.lower_bound !== 0);
-          // Re-simulate the model with the updated bounds
-          // adding first all the modifications from the design
-          // and then the modifications (edited bounds) computed above
-          this.postSimulation(card, model, [
-            ...this.cardModifications(card),
-            ...editedBounds
-          ]);
-        })
-        .catch(error => {
-          this.updateCard({
-            uuid: card.uuid,
-            props: { hasSimulationError: true }
-          });
-          this.hasSimulationError = true;
-        })
-        .then(response => {
-          this.updateCard({
-            uuid: card.uuid,
-            props: { isSimulating: false }
-          });
-        });
+            }
+            if (newBound < 0) {
+              return {
+                operation: "modify",
+                type: "reaction",
+                id: manipulation.id,
+                data: {
+                  upper_bound: newBound,
+                  lower_bound: model_reaction.lower_bound
+                }
+              };
+            }} else if (manipulation.direction === "down") {
+            if (newBound > 0) {
+              return {
+                operation: "modify",
+                type: "reaction",
+                id: manipulation.id,
+                data: {
+                  upper_bound: newBound,
+                  lower_bound: model_reaction.lower_bound
+                }
+              };
+            }
+            if (newBound < 0) {
+              return {
+                operation: "modify",
+                type: "reaction",
+                id: manipulation.id,
+                data: {
+                  lower_bound: newBound,
+                  upper_bound: model_reaction.upper_bound
+                }
+              };
+            }
+            } else if (manipulation.direction === "invert") {
+            if (newBound > 0) {
+              return {
+                operation: "modify",
+                type: "reaction",
+                id: manipulation.id,
+                data: {
+                  lower_bound: newBound,
+                  upper_bound: model_reaction.upper_bound
+                }
+              };
+            }
+            if (newBound < 0) {
+              return {
+                operation: "modify",
+                type: "reaction",
+                id: manipulation.id,
+                data: {
+                  upper_bound: newBound,
+                  lower_bound: model_reaction.lower_bound
+                }
```

Comment:
Would all those combinations be more readable like this?
```ts
const editedBounds = card.manipulations.map(manipulation => {
  const model_reaction = model.model_serialized.reactions.find(
    rxn => rxn.id === manipulation.id
  );
  const oldLowerBound = model_reaction.lower_bound;
  const oldUpperBound = model_reaction.upper_bound;

  // manipulation.value can never be equal to zero hence we don't need to check for it.
  const newBound = manipulation.value;

  if (
    (manipulation.direction === "up" && newBound > 0) ||
    (manipulation.direction === "invert" && newBound > 0) ||
    (manipulation.direction === "down" && newBound < 0)
  ) {
    return {
      operation: "modify",
      type: "reaction",
      id: manipulation.id,
      data: {
        lower_bound: newBound,
        upper_bound: oldUpperBound
      }
    };
  }

  if (
    (manipulation.direction === "up" && newBound < 0) ||
    (manipulation.direction === "invert" && newBound < 0) ||
    (manipulation.direction === "down" && newBound > 0)
  ) {
    return {
      operation: "modify",
      type: "reaction",
      id: manipulation.id,
      data: {
        lower_bound: oldLowerBound,
        upper_bound: newBound
      }
    };
  }
})

```

Reply:
Certainly would be :O